app/src/main/java/com/projectfirebase/soen341/root/signUpActivity.java

```diff
@@ -48,7 +62,43 @@ public void onClick(View view) {
         });
     }
 
-    private void signUp(){
+    private void signUp() {
+        final String email = editTextSUEmail.getText().toString().trim();
+        final String firstName = editTextSUFirstName.getText().toString().trim();
+        final String lastName = editTextSULastName.getText().toString().trim();
+        final String phoneNumber = editTextSUPhone.getText().toString().trim();
+        final String ZIPCode = editTextSUZIP.getText().toString().trim();
+
+        final String password = editTextSUPassword.getText().toString().trim();
+        final String confPassword = editTextSUConfPassword.getText().toString().trim();
+
+    // Validate all fields
+        if (firstName.isEmpty() || lastName.isEmpty() || phoneNumber.isEmpty() || ZIPCode.isEmpty()) {
+            Toast.makeText(signUpActivity.this, "Please fill all fields.", Toast.LENGTH_SHORT).show();
+        } else if (!password.equals(confPassword)) {
+            Toast.makeText(signUpActivity.this, "Passwords do not match.", Toast.LENGTH_SHORT).show();
+        } else {
+
+        // Create user
+            authRef.createUserWithEmailAndPassword(email, password).addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+                @Override
+                public void onComplete(@NonNull Task<AuthResult> task) {
+                    if (task.isSuccessful()) {
+                        Toast.makeText(signUpActivity.this, "Sign up successful!", Toast.LENGTH_LONG).show();
+                        FirebaseUser user = task.getResult().getUser();
+                        String UID = user.getUid();
+                        User newUser = new User(firstName, lastName, phoneNumber, ZIPCode);
+                        usersRef.child(UID).setValue(newUser);
+                    } else {
+                        try {
+                            throw task.getException();
+                        } catch (Exception e) {
+                            Toast.makeText(signUpActivity.this, task.getException().getMessage(), Toast.LENGTH_LONG).show();
```

Comment:
Is it a good idea to show the end user the actual exception error, or should we just show them something like "Sorry, we are having difficulty processing your request. Please try again."?

Reply:
For now we'll leave it as is for testing if anything goes wrong with it. We can standardize our error messages in the future in string resource.

Reply:
I think we could put both since the message is important for them but the exception message can be useful for technical support. Also we can link this the bug report.